### ISSUE TITLE
Navigator.NavigateTo wont work directly after adding the tile layer (Forms)

### ISSUE THREAD

**Issue by Sebastian1989101** [Mapsui/Mapsui#573]
The code below should explain itself. The equivalent in mapsui 1.3.x works fine but it seems like the behavior differs in 2.0.0.

```
var tileLayer = new TileLayer(new MbTilesTileSource(new SQLiteConnectionString(tileDatabase, true, null)));
tileLayer.Attribution = null;
tileLayer.MemoryCache.MaxTiles = 128;
tileLayer.MemoryCache.MinTiles = 64;

Map.Layers.Add(tileLayer);

var mapResolutions = Map.Resolutions;
//does not work, except I wait before this call a couple of seconds
Navigator.NavigateTo(new Point(0, 0), mapResolutions[mapResolutions.Length / 2]); 
```

**Comment by charlenni** [comment]
Do you get an error?

**Comment by pauldendulk** [comment]
NavigateTo does not work during initialization because the map Width/Height needs to be known. Is this code called during initialization? To initialize the Map.Home method should be used. 

I understand this is confusing but it is hard to circumvent because it is logically not possible to determine a resolution if you don't know the map size. Perhaps we should store the NavigateTo command and execute if once the Width/Height is known. There might be other options, like storing this information in a different way. In the current implementation the viewport stores a resolution and a center.

**Comment by Sebastian1989101** [comment]
@charlenni No I don't think so. This is the whole log from the part above: 
```
[0:] ######################### START CHANING TILE DATABASE #########################
01-07 17:58:53.464 D/Mono    (14507): Unloading image System.Linq.dll [0x889af600].
01-07 17:58:53.464 D/Mono    (14507): Image addref System.Linq[0x86fcfa20] -> System.Linq.dll[0x89932c00]: 6
01-07 17:58:53.464 D/Mono    (14507): Config attempting to parse: 'System.Linq.dll.config'.
01-07 17:58:53.464 D/Mono    (14507): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d15-9/xamarin-android/external/mono/sdks/out/android-x86-release/etc/mono/assemblies/System.Linq/System.Linq.config'.
01-07 17:58:53.464 D/Mono    (14507): Assembly Ref addref BruTile.MbTiles[0x9db1d200] -> System.Linq[0x9d629ac0]: 4
01-07 17:58:53.481 D/Mono    (14507): DllImport searching in: 'e_sqlite3' ('libe_sqlite3.so').
01-07 17:58:53.481 D/Mono    (14507): Searching for 'sqlite3_bind_text'.
01-07 17:58:53.488 D/Mono    (14507): Unloading image System.Globalization.dll [0x889af100].
01-07 17:58:53.488 D/Mono    (14507): Image addref System.Globalization[0x86fcfcc0] -> System.Globalization.dll[0x8956d000]: 6
01-07 17:58:53.488 D/Mono    (14507): Config attempting to parse: 'System.Globalization.dll.config'.
01-07 17:58:53.488 D/Mono    (14507): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d15-9/xamarin-android/external/mono/sdks/out/android-x86-release/etc/mono/assemblies/System.Globalization/System.Globalization.config'.
01-07 17:58:53.488 D/Mono    (14507): Assembly Ref addref BruTile.MbTiles[0x9db1d200] -> System.Globalization[0x9d629a60]: 4
01-07 17:58:53.498 D/Mono    (14507): Unloading image System.Runtime.Extensions.dll [0x889afb00].
01-07 17:58:53.499 D/Mono    (14507): Image addref System.Runtime.Extensions[0x86fcfd20] -> System.Runtime.Extensions.dll[0x89932700]: 6
01-07 17:58:53.499 D/Mono    (14507): Config attempting to parse: 'System.Runtime.Extensions.dll.config'.
01-07 17:58:53.499 D/Mono    (14507): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d15-9/xamarin-android/external/mono/sdks/out/android-x86-release/etc/mono/assemblies/System.Runtime.Extensions/System.Runtime.Extensions.config'.
01-07 17:58:53.499 D/Mono    (14507): Assembly Ref addref BruTile.MbTiles[0x9db1d200] -> System.Runtime.Extensions[0x9d629b20]: 4
01-07 17:58:53.593 D/Mono    (14507): Unloading image System.Runtime.Extensions.dll [0x889af100].
01-07 17:58:53.593 D/Mono    (14507): Image addref System.Runtime.Extensions[0x86fcff00] -> System.Runtime.Extensions.dll[0x89932700]: 7
01-07 17:58:53.593 D/Mono    (14507): Config attempting to parse: 'System.Runtime.Extensions.dll.config'.
01-07 17:58:53.594 D/Mono    (14507): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d15-9/xamarin-android/external/mono/sdks/out/android-x86-release/etc/mono/assemblies/System.Runtime.Extensions/System.Runtime.Extensions.config'.
01-07 17:58:53.594 D/Mono    (14507): Assembly Ref addref BruTile[0x9db1d140] -> System.Runtime.Extensions[0x9d629b20]: 5
01-07 17:58:53.656 D/Mono    (14507): Unloading image System.Threading.dll [0x889afb00].
01-07 17:58:53.656 D/Mono    (14507): Image addref System.Threading[0x86fcfa80] -> System.Threading.dll[0x89570c00]: 6
01-07 17:58:53.656 D/Mono    (14507): Config attempting to parse: 'System.Threading.dll.config'.
01-07 17:58:53.656 D/Mono    (14507): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d15-9/xamarin-android/external/mono/sdks/out/android-x86-release/etc/mono/assemblies/System.Threading/System.Threading.config'.
01-07 17:58:53.656 D/Mono    (14507): Assembly Ref addref BruTile[0x9db1d140] -> System.Threading[0x894f06c0]: 4
[0:] ######################### END CHANING TILE DATABASE #########################
```

@pauldendulk Nothing happens if I use panning as input after the NavigateTo. I guess the call to NavigateTo just ends in nothing.

**Comment by charlenni** [comment]
@Sebastian1989101 I checked this and couldn't find any problem. I used your project, which you provided at #570. Than I added the ````NavigateTo()````, which didn't cause any problems. Perhaps there is something else in the code?

**Comment by charlenni** [comment]
Btw. tested it on iPhone simulator. Perhaps you used any other platform?

**Comment by Sebastian1989101** [comment]
@charlenni I used an Android 9 Simulator for this because of the other issue I had with iOS.

**Comment by pauldendulk** [comment]
Could you test this with 2.0.0-beta.14?

**Comment by Sebastian1989101** [comment]
@pauldendulk At least in the Android 9 simulator nothing changed and it still won't work directly after adding the layer. I will test it further in the next days but at first look, it seems to be the same.

**Comment by pauldendulk** [comment]
Thanks for testing. I will investigate it further.

**Comment by Sebastian1989101** [comment]
I just did a quick test on the iOS simlator too and get the same result as on Android. Without waiting for the map to be fully initzialized it wont react to NavigateTo(). Also it seems like it needs a bit more time to pop in the info layers (~1200 icons).

**Comment by pauldendulk** [comment]
Trying to reproduce this. Failed so far on WPF.

Before you add the mbtiles layer there is no layer at all?

**Comment by pauldendulk** [comment]
I created a branch to reproduce it but failed so far
https://github.com/Mapsui/Mapsui/tree/navigateto-wont-work

**Comment by pauldendulk** [comment]
This is how I try to reproduce it
- Start Mapsui.Samples.Forms.Android
- Select OpenStreetMap. I have removed the OpenStreetMap layer, it is now an empty map.
- Press the 'DO IT' button at the top.
- The map shows. I used your mbtiles database.

Here is the code under the button:
https://github.com/Mapsui/Mapsui/blob/navigateto-wont-work/Samples/Mapsui.Samples.Forms/Mapsui.Samples.Forms.Shared/MapPage.xaml.cs#L136
